packages/transactions/src/builders.ts

```diff
@@ -1692,3 +1663,30 @@ export async function estimateTransactionFeeWithFallback(
     throw error;
   }
 }
+
+/** @internal multi-sig signing re-use */
+function mutatingSignMultiSig(
+  /** **Warning:** method mutates `transaction` */
+  transaction: StacksTransaction,
+  publicKeys: string[],
+  signerKeys: string[]
+) {
+  const signer = new TransactionSigner(transaction);
+
+  // sort public keys (only for newer non-sequential multi-sig for backwards compatibility)
+  if (isNonSequential(transaction.auth.spendingCondition.hashMode)) {
+    publicKeys.sort();
+  }
+
+  // sign in order of public keys
+  for (const publicKey of publicKeys) {
+    const signerKey = signerKeys.find(key => bytesToHex(pubKeyfromPrivKey(key).data) === publicKey);
+    if (signerKey) {
+      // either sign and append message signature (which allows for recovering the public key)
+      signer.signOrigin(createStacksPrivateKey(signerKey));
+    } else {
+      // or append the public key (which did not sign here)
+      signer.appendOrigin(publicKeyFromBytes(hexToBytes(publicKey)));
+    }
+  }
+}
```

Comment:
@jbencin you'll notice this now differs from the blockchain `make` methods, but I think is more correct. We try to sign/append in order of public-keys, rather than first all signs and then all appends _(not exactly sure why that was in the code previously, but seems to be inspired by stacks-core methods)_